src/SampleWebApp/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/SampleWebAppDbContext.g.cs

```diff
@@ -42,20 +42,34 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         }
     }
     
+    public static Type? GetTypeByEntityName(string entityName)
+    {
+        return Type.GetType("SampleWebApp.Domain." + entityName);
+    }
+    
     protected override void OnModelCreating(ModelBuilder modelBuilder)
     {
         base.OnModelCreating(modelBuilder);
         if (_noxSolution.Domain != null)
         {
             var codeGeneratorState = new NoxSolutionCodeGeneratorState(_noxSolution);
             
+            var relationships = ((INoxDatabaseConfigurator)_dbProvider).GetRelationshipsToCreate(codeGeneratorState, _noxSolution.Domain.Entities, modelBuilder);
+            relationships = relationships
+                .OrderBy(x => x.Entity.Name)
+                .ToList();
+            
+            foreach (var relationship in relationships)
+            {
+                relationship.RelationshipEntityType = GetTypeByEntityName(relationship.Relationship.Entity)!;
+            }
+            
             foreach (var entity in _noxSolution.Domain.Entities)
             {
-                var type = Type.GetType("SampleWebApp.Domain." + entity.Name);
-                
+                var type = GetTypeByEntityName(entity.Name);
```

Comment:
if you want to encapsulate this logic, I would consider the codeGeneratorState to do this